### PR TITLE
Remove restrictions on external prefix redirects

### PR DESCRIPTION
**Pull request by cbaines**
This is to allow for redirecting trade tariff requests to the service
domain.

### CODE REVIEW

**Comment by cbaines** [comment]
I struggled loads with separating out the whitelist behaviour and replacing it with checking external domains are within `.gov.uk`. I eventually ended up changing most of the validation, in a similar way to the changes I made to the validation inside the Publishing API.

There are some interesting differences between the validation here and the Publishing API, e.g. the Publishing API doesn't do any validation depending on the segments mode, whereas that is done here.

File: app/models/route.rb [hunk 1]
```diff
@@ -84,62 +83,66 @@ def default_segments_mode
...
+    end
   rescue URI::InvalidURIError
-    false
+    errors[:redirect_to] << "is an invalid URI"
   end
 
-  def path_is_root_or_empty?(uri)
-    uri.path.blank? || uri.path == '/'
-  end
+  def validate_external_target?(target)
```
Comment (danielroseman) [inline app/models/route.rb]: This method and the one following shouldn't end with `?`; they don't return whether the target is valid (in which case they would be called `valid_internal_target?`, they actually do the validation.

File: spec/models/route_spec.rb [hunk 2]
```diff
@@ -205,7 +205,7 @@
           end
 
           it "will allow external URLs" do
-            route.redirect_to = "http://example.com/thing"
+            route.redirect_to = "http://example.service.gov.uk/thing"
```
Comment (danielroseman) [inline spec/models/route_spec.rb]: We should test that it rejects non-gov.uk domains.

**Comment by cbaines** [comment]
> Where is this validation surfaced?

As you say below, at the moment, the only things to my knowledge talking to the router-api are the content stores, and router-data. Both of them might do something with errors, as I assume they are returned when requests are made over HTTP.

> Generally the only thing that is talking to this app is the content-store, which itself gets the data from publishing-api so it should already be validated there. Would this only be used for router-data?

External redirects to campaigns are currently managed through router-data [1], and I'm guessing that this is what led to there being some validation on redirect destinations here.

1: Caveat here is that the Content Tools team did some work over the Blitz period to support redirects to .gov.uk subdomains to go through the Publishing API, so I think at least one campaign redirect is done that way now, and there is nothing to my knowledge blocking the others from being migrated.

File: app/models/route.rb (hunk 1 above)
Comment (cbaines) [inline app/models/route.rb]: Good spot, that tripped me up as well, but I didn't change the name. I've done that now.

File: spec/models/route_spec.rb (hunk 2 above)
Comment (cbaines) [inline spec/models/route_spec.rb]: I've now added this in 6d33f81.

File: spec/models/route_spec.rb [hunk 3]
```diff
@@ -53,7 +53,7 @@
         ].each do |path|
           route.incoming_path = path
           expect(route).not_to be_valid
-          expect(route.errors[:incoming_path].size).to eq(1)
+          expect(route.errors[:incoming_path].size).not_to eq(0)
```
Comment (cbaines) [inline spec/models/route_spec.rb]: Thanks, I've changed that now :)

**Comment by cbaines** [comment]
Thanks for your review Daniel :+1:

**Pull request merged**